smart-connector/src/test/java/eu/knowledge/engine/smartconnector/misc/ConfigurationTest.java

```diff
@@ -0,0 +1,212 @@
+import eu.knowledge.engine.smartconnector.api.Binding;
+import eu.knowledge.engine.smartconnector.api.BindingSet;
+import eu.knowledge.engine.smartconnector.api.CommunicativeAct;
+import eu.knowledge.engine.smartconnector.api.ExchangeInfo;
+import eu.knowledge.engine.smartconnector.api.GraphPattern;
+import eu.knowledge.engine.smartconnector.impl.SmartConnectorConfig;
+import eu.knowledge.engine.smartconnector.util.KnowledgeNetwork;
+import eu.knowledge.engine.smartconnector.util.MockedKnowledgeBase;
+
+public class ConfigurationTest {
+
+	private static final Logger LOG = LoggerFactory.getLogger(ConfigurationTest.class);
+
+	private KnowledgeNetwork kn;
+	private MockedKnowledgeBase kb1;
+	private AskKnowledgeInteraction askKI;
+	private MockedKnowledgeBase kb2;
+	private AnswerKnowledgeInteraction answerKI;
+	private int waitTimeout = 0;
+
+	@BeforeEach
+	public void beforeTest() {
+		this.kn = new KnowledgeNetwork();
+
+		intializeKB1();
+		intializeKB2();
+		kn.addKB(kb1);
+		kn.addKB(kb2);
+		kn.sync();
+	}
+
+	@Test
+	public void testConfigValidateTrue() {
+		System.setProperty(SmartConnectorConfig.CONF_KEY_VALIDATE_OUTGOING_BINDINGS_WRT_INCOMING_BINDINGS, "true");
+		BindingSet bs = new BindingSet();
+
+		var bs1 = new Binding();
+		bs1.put("s", "<barry2>");
+		bs.add(bs1);
+
+		var future = kb1.ask(this.askKI, bs);
+
+		AskResult askResult = null;
+		try {
+			askResult = future.get();
+			assertTrue(askResult.getExchangeInfoPerKnowledgeBase().iterator().hasNext());
+			var info = askResult.getExchangeInfoPerKnowledgeBase().iterator().next();
+
+			assertEquals(ExchangeInfo.Status.FAILED, info.getStatus());
+			assertTrue(info.getFailedMessage().contains("java.lang.IllegalArgumentException"));
+
+		} catch (InterruptedException | ExecutionException e) {
+			LOG.info("{}", e);
+		}
+
+		LOG.info("Result: {}", askResult);
+
+		System.clearProperty(SmartConnectorConfig.CONF_KEY_VALIDATE_OUTGOING_BINDINGS_WRT_INCOMING_BINDINGS);
+	}
+
+	@Test
+	public void testConfigValidateFalse() {
+		System.setProperty(SmartConnectorConfig.CONF_KEY_VALIDATE_OUTGOING_BINDINGS_WRT_INCOMING_BINDINGS, "false");
+
+		BindingSet bs = new BindingSet();
+
+		var bs1 = new Binding();
+		bs1.put("s", "<barry2>");
+		bs.add(bs1);
+
+		var future = kb1.ask(this.askKI, bs);
+
+		AskResult askResult = null;
+		try {
+			askResult = future.get();
+			assertTrue(askResult.getExchangeInfoPerKnowledgeBase().iterator().hasNext());
+			var info = askResult.getExchangeInfoPerKnowledgeBase().iterator().next();
+
+			assertEquals(ExchangeInfo.Status.SUCCEEDED, info.getStatus());
+			assertEquals(null, info.getFailedMessage());
+
+		} catch (InterruptedException | ExecutionException e) {
+			LOG.info("{}", e);
+		}
+
+		LOG.info("Result: {}", askResult);
+		System.clearProperty(SmartConnectorConfig.CONF_KEY_VALIDATE_OUTGOING_BINDINGS_WRT_INCOMING_BINDINGS);
+	}
+
+	@Test
+	public void testConfigWaitForKnowledgeBaseNegative() {

```

Comment:
What do you mean with "Negative" in this test name? It doesn't seem like you test a negative wait time.

Reply:
I meant 'negative' in the sense that the ASK is expected to fail and return an `TimeoutException`. See line [129](https://github.com/TNO/knowledge-engine/pull/569/files/4c9f67a4d6215c5ad17c416b85346781e26a4b64#diff-9291acd974b63cdc62fd32026d76fd0f72d03cf28aedca28abd3bd8f1bd744c6R129).